### PR TITLE
feat(python): Add functionality and resiliency to wmill python client

### PR DESCRIPTION
**Pull request by knowsuchagency**
This script updates the `wmill` client to add additional arguments to the methods for running scripts synchronously.

* `assert_result_is_not_none`
* `cleanup`
* `timeout`

`assert_result_is_not_none` asserts that the result of the script isn't null. The current behavior is to raise an exception if the result is falsey, which causes unnecessary failures if the result is something like 0, probably a bug.

`cleanup` ensures that jobs that are cancelled gracefully when the parent script is cancelled.

`timeout` throws a `TimeoutError` if passed and the script takes longer than it to execute

### CODE REVIEW

**Comment by knowsuchagency** [comment]
I have read the CLA Document and I hereby sign the CLA

File: python-client/wmill/wmill/client.py [hunk 1]
```diff
@@ -81,21 +87,71 @@ def run_script_async(
...
     """
+    args = args or {}
     job_id = run_script_async(hash, args, None)
+
+    def cancel_job():
+        from windmill_api.api.job.cancel_queued_job import (
+            sync_detailed,
+            CancelQueuedJobJsonBody,
+        )
+        import wmill
```
Comment (rubenfiszel) [inline python-client/wmill/wmill/client.py]: I think this can be removed

File: python-client/wmill/wmill/client.py [hunk 2]
```diff
@@ -81,21 +87,71 @@ def run_script_async(
     ).content.decode("us-ascii")
 
 
-def run_script_sync(hash: str, args: Dict[str, Any] = {}, verbose: bool = False) -> Dict[str, Any]:
+def run_script_sync(
+    hash: str, 
+    args: Optional[Dict[str, Any]] = None, 
+    verbose: bool = False,
+    assert_result_is_not_none: bool = True,
+    cancel_atexit: bool = True,
```
Comment (rubenfiszel) [inline python-client/wmill/wmill/client.py]: cancel_at_exit*

I had a hard time reading atexit :)

**Review by rubenfiszel — DISMISSED** [review]
Very cool, only minor changes needed

File: python-client/wmill/wmill/client.py (hunk 2 above)
Comment (knowsuchagency) [reply python-client/wmill/wmill/client.py]: I'll rename it :)

**Pull request merged**